components/BarcodeScanner.vue

```diff
@@ -0,0 +1,334 @@
+            },
+            facingMode: {
+              ideal: 'environment',
+            }
+          },
+        });
+
+        this.scanner.onUnduplicatedRead = txt => {
+          this.$emit('result', this.getDLInfo(txt));
+        };
+
+        await this.scanner.setUIElement(this.$refs.videoContainer);
+
+        await this.scanner.open();
+      } catch(e) {
+        console.error(e);
+        this.$emit('error', e.message);
+      } finally {
+        this.loadingCamera = false;
+      }
+    },
+    getDLInfo(txt) {
+      const lines = txt.split('\n');
+      const abbrs = Object.keys(driverLicenseFields);
+      const dlInfo = {};
+      lines.forEach((line, i) => {
+        let abbr;
+        let content;
+        if(i === 1) {
+          abbr = 'DAQ';
+          content = line.substring(line.indexOf(abbr) + 3);
+        } else {
+          abbr = line.substring(0, 3);
+          content = line.substring(3).trim();
+        }
+        if(abbrs.includes(abbr)) {
+          dlInfo[abbr] = {
+            description: driverLicenseFields[abbr],
+            value: content,
+          };
+        }
+      });
+      return dlInfo;
+    },
+    cvsDrawArea() {
+      this.maskCanvas.width = this.clientWidth;
+      this.maskCanvas.height = this.clientHeight;
+      this.maskCanvasContext.fillStyle = 'rgba(0,0,0,.5)';
+      this.maskCanvasContext.fillRect(
+        0,
+        0,
+        this.maskCanvas.width,
+        this.maskCanvas.height
+      );
+      this.maskCanvasContext.globalCompositeOperation = 'destination-out';
+      this.maskCanvasContext.fillStyle = 'white';
+      this.maskCanvasContext.fillRect(
+        (this.regionLeft / 100) * this.maskCanvas.width,
+        (this.regionTop / 100) * this.maskCanvas.height,
+        ((100 - 2 * this.regionLeft) / 100) * this.maskCanvas.width,
+        ((100 - 2 * this.regionTop) / 100) * this.maskCanvas.height
+      );
+      this.maskCanvasContext.globalCompositeOperation = 'source-over';
+      this.maskCanvasContext.lineWidth = 2;
+      this.maskCanvasContext.strokeStyle = this.guideColor;
+      this.maskCanvasContext.rect(
+        (this.regionLeft / 100) * this.maskCanvas.width,
+        (this.regionTop / 100) * this.maskCanvas.height,
+        ((100 - 2 * this.regionLeft) / 100) * this.maskCanvas.width,
+        ((100 - 2 * this.regionTop) / 100) * this.maskCanvas.height
+      );
+      this.maskCanvasContext.stroke();
+    },
+    async updateCamera(deviceId) {
+      if(this.currentCamera.deviceId === deviceId) {
+        return;
+      }
+      this.loading = true;
+      await this.scanner.setCurrentCamera(deviceId);
+      this.currentCamera = await this.scanner.getCurrentCamera();
+      this.loading = false;
+    },
+    async upload(event) {
+      await this.imageScan();
+
+      const files = event.target.files;
+      // no file selected
+      if(files.length === 0) {
+        return;
+      }
+      for(let i = 0; i < files.length; ++i) {
```

Comment:
This implementation is going to cause the main loop to try and process all files concurrently. I'll file an issue to resolve later.

Reply:
#10